### ISSUE TITLE
[clang-format] incorrect space added r-value reference 

### ISSUE THREAD

**Issue by tdegeus** [llvm/llvm-project#71681]
```cpp
struct foo {
  template <class T> decltype(auto) operator()(T&& t) const {}
}
```
should not be formatted with
```
PointerBindsToType: true
```
but is formatted to
```cpp
struct foo {
  template <class T> decltype(auto) operator()(T && t) const {}
}
```

clang-format 17.0.4

**Comment by rymiel** [comment]
This seems to have regressed in 3f3620e5c9ee0f7b64afc39e5a26c6f4cc5e7b37, but was then fixed by 67b99fa8ba638edddd5ce38b4d379eab8e339e4a on main, but that fix is not in a release yet
Duplicate of #66890

**Comment by owenca** [comment]
67b99fa8ba63 has been [backported](https://github.com/llvm/llvm-project-release-prs/pull/770) just now.
